File: imacals-dashboard/src/services/organization.ts
import { api } from '@/services/api';

export interface Org {
  id: string;
  name: string;
  slug: string;
  parent_id: string | null;
  description: string | null;
}

export const organizationService = {
  index: (): Promise<Org[]> => api.get<Org[]>('/organizations'),
};

File: imacals-api/src/models/organization.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;
use crate::models::permission::Permission;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Organization {
    pub id: Uuid,
    pub name: String,
    pub parent_id: Option<Uuid>,
    pub description: Option<String>,
    pub slug: String,
    pub created_by: Uuid,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub deleted_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrganizationWithPermissions {
    #[serde(flatten)]
    pub organization: Organization,
    pub permissions: Vec<Permission>,
}

impl Organization {
    pub fn is_imacals(&self) -> bool {
        self.slug == "imacals"
    }
}
